include/crocoddyl_msgs/inertial_parameters_subscriber.h

```diff
@@ -0,0 +1,156 @@
+//
+// Copyright (C) 2020-2023, Heriot-Watt University, University of Oxford
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_SUBSCRIBER_H_
+#define CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_SUBSCRIBER_H_
+
+#include "crocoddyl_msgs/MultibodyInertialParameters.h"
+
+#include <mutex>
+#ifdef ROS2
+#include <rclcpp/rclcpp.hpp>
+#else
+#include <ros/node_handle.h>
+#endif
+
+typedef crocoddyl_msgs::MultibodyInertialParameters MultibodyInertialParameters;
+typedef const crocoddyl_msgs::MultibodyInertialParameters::ConstPtr
+    &MultibodyInertialParametersSharedPtr;
+typedef Eigen::Matrix<double, 10, 1> Vector10d;
+namespace crocoddyl_msgs {
+
+
+class MultibodyInertialParametersRosSubscriber {
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+   /**
+   * @brief Initialize the multi-body inertial parameters subscriber.
+   *
+   * @param[in] topic     Topic name
+   */
+  MultibodyInertialParametersRosSubscriber(
+      const std::string &topic = "/crocoddyl/inertial_parameters"):
+#ifdef ROS2
+      : node_(rclcpp::Node::make_shared("inertia_parameters_subscriber")),
+        sub_(node_->create_subscription<MultibodyInertialParameters>(
+            topic, 1,
+            std::bind(&MultibodyInertialParametersRosSubscriber::callback, this,
+                      std::placeholders::_1))),
+        has_new_msg_(false), is_processing_msg_(false), last_msg_time_(0.) {
+    spinner_.add_node(node_);
+    thread_ = std::thread([this]() { this->spin(); });
+    thread_.detach();
+    RCLCPP_INFO_STREAM(node_->get_logger(),
+                       "Subscribing MultibodyInertialParameters messages on " << topic);
+#else 
+      spinner_(2), has_new_msg_(false), is_processing_msg_(false), last_msg_time_(0.) {
+    ros::NodeHandle n;
+    sub_ = n.subscribe<MultibodyInertialParameters>(
+        topic, 1, &MultibodyInertialParametersRosSubscriber::callback, this,
+        ros::TransportHints().tcpNoDelay());
+    spinner_.start();
+    ROS_INFO_STREAM("Subscribing MultibodyInertialParameters messages on " << topic);
+#endif
+  }
+
+  ~MultibodyInertialParametersRosSubscriber() = default;
+
+  /**
+   * @brief Get the latest inertial parameters
+   *
+   * @return  A map from body names to inertial parameters.
+   */
+  
+  std::map<std::string, Vector10d>
+  get_parameters() {
+    // start processing the message
+    is_processing_msg_ = true;
+    std::lock_guard<std::mutex> guard(mutex_);
+    
+    const std::size_t n_bodies = msg_.parameters.size();
+    for (std::size_t i = 0; i<n_bodies; ++i){
+      const std::string &name =  msg_.parameters[i].name;
+      p_tmp_[0] = msg_.parameters[i].inertia.m;
+      p_tmp_[1] = msg_.parameters[i].inertia.com.x;
+      p_tmp_[2] = msg_.parameters[i].inertia.com.y;
+      p_tmp_[3] = msg_.parameters[i].inertia.com.z;
```

Comment:
This doesn't follow the Pinocchio convention. We should do:
```suggestion
      p_tmp_[1] = p_tmp_[0] * msg_.parameters[i].inertia.com.x;
      p_tmp_[2] = p_tmp_[0] * msg_.parameters[i].inertia.com.y;
      p_tmp_[3] = p_tmp_[0] * msg_.parameters[i].inertia.com.z;
```

Additionally, I would rename `p_tmp_` by `psi_tmp_` to follow a unique convention in our code.